src/Opdex.Platform.Infrastructure/Data/Handlers/Transactions/SelectTransactionForVaultProposalVoteRewindQueryHandler.cs

```diff
@@ -0,0 +1,101 @@
+using AutoMapper;
+using MediatR;
+using Opdex.Platform.Common.Models;
+using Opdex.Platform.Domain.Models.TransactionLogs;
+using Opdex.Platform.Domain.Models.Transactions;
+using Opdex.Platform.Infrastructure.Abstractions.Data;
+using Opdex.Platform.Infrastructure.Abstractions.Data.Extensions;
+using Opdex.Platform.Infrastructure.Abstractions.Data.Models.Transactions;
+using Opdex.Platform.Infrastructure.Abstractions.Data.Models.Transactions.TransactionLogs;
+using Opdex.Platform.Infrastructure.Abstractions.Data.Queries.Transactions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Opdex.Platform.Infrastructure.Data.Handlers.Transactions;
+
+public class SelectTransactionForVaultProposalVoteRewindQueryHandler
+    : IRequestHandler<SelectTransactionForVaultProposalVoteRewindQuery, Transaction>
+{
+    private const string Split = "Split";
+
+    private static readonly string SqlQuery =
+        @$"SELECT
+                t.{nameof(TransactionEntity.Id)},
+                t.`{nameof(TransactionEntity.To)}`,
+                t.`{nameof(TransactionEntity.From)}`,
+                t.{nameof(TransactionEntity.Hash)},
+                t.{nameof(TransactionEntity.Block)},
+                t.{nameof(TransactionEntity.GasUsed)},
+                t.{nameof(TransactionEntity.NewContractAddress)},
+                t.{nameof(TransactionEntity.Success)},
+                true as {Split},
+                tl.{nameof(TransactionLogEntity.Id)},
+                tl.{nameof(TransactionLogEntity.TransactionId)},
+                tl.{nameof(TransactionLogEntity.LogTypeId)},
+                tl.{nameof(TransactionLogEntity.SortOrder)},
+                tl.{nameof(TransactionLogEntity.Contract)},
+                tl.{nameof(TransactionLogEntity.Details)}
+            FROM transaction t
+            LEFT JOIN transaction_log tl
+                ON t.{nameof(TransactionEntity.Id)} = tl.{nameof(TransactionLogEntity.TransactionId)}
+            WHERE t.`{nameof(TransactionEntity.From)}` = @{nameof(SqlParams.Voter)} AND
+                  tl.{nameof(TransactionLogEntity.Contract)} = @{nameof(SqlParams.Vault)} AND
+                  tl.{nameof(TransactionLogEntity.LogTypeId)} IN @{nameof(SqlParams.LogTypes)} AND
+                  JSON_EXTRACT(tl.{nameof(TransactionLogEntity.Details)}, '$.proposalId') = @{nameof(SqlParams.ProposalId)}
+            ORDER BY t.{nameof(TransactionEntity.Block)} DESC;".RemoveExcessWhitespace();
```

Comment:
Is it possible to instead have this as two separate queries? One to retrieve the transaction then one to retrieve the log. Might improve performance.

Reply:
It'll end up being the same query because we have to filter by log and proposalId anyways, might as well use Dapper to build in one, due to the join Limit 1 won't be exact since it would pull one log for a transaction technically. I can set a lower limit like 5 or so to help out. 

I didn't enjoy making this query but its needed because the contract only stores the active balance of a pledge/vote. It doesn't keep the official vote/pledge total for a user in State, only logs.